test/AST/FunctionDecl.cpp

```diff
@@ -0,0 +1,48 @@
+#include "FunctionDecl.hpp"
+
+#include <gtest/gtest.h>
+
+using namespace glu::ast;
+
+class FunctionDeclTest : public ::testing::Test {
+protected:
+    glu::SourceLocation loc;
+    FunctionDeclTest() : loc(11) { }
+};
+
+class TestDeclBase : public DeclBase {
+public:
+    TestDeclBase(NodeKind kind, glu::SourceLocation loc, ASTNode *parent)
+        : DeclBase(kind, loc, parent)
+    {
+    }
+};
+
+TEST_F(FunctionDeclTest, FunctionDeclConstructor)
+{
+    std::string name = "foo";
+    std::vector<glu::types::TypeBase *> parameters
+        = { new glu::types::BoolTy(), new glu::types::BoolTy() };
+    glu::types::TypeBase *returnType = new glu::types::BoolTy();
+
+    glu::types::FunctionTy type(parameters, returnType);
+    llvm::SmallVector<Param> params = { Param("a", new glu::types::BoolTy()),
+                                        Param("b", new glu::types::BoolTy()) };
```

Comment:
don't recreate new BoolTy every time, use the same instance (in the real implementation, all types will be uniqued)